version: "3.7"

services:
  nginx:
    image: nginx:1.19.5
    ports:
      - 80:80
    networks:
      - dabin_network
    volumes:
      - /home/kangdb123-/nginx.conf:/etc/nginx/nginx.conf
      - dabin_static:/data/static/
      - dabin_media:/data/media/

  django:
    image: dabin_django:4
    networks:
      - dabin_network
    volumes:
      - dabin_static:/home/gis_web_mine/staticfiles/
      - dabin_media:/home/gis_web_mine/media/
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD

  mariadb:
    image: mariadb:10.5
    networks:
      - dabin_network
    volumes:
      - gis_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_ROOT_PASSWORD
      - MARIADB_USER
      - MARIADB_PASSWORD
    environment:
      MARIADB_ROOT_PASSWORD: password
      MARIADB_DATABASE: django
      MARIADB_USER: django
      MARIADB_PASSWORD: pass


networks:
  dabin_network:

volumes:
  dabin_static:
  dabin_media:
  gis_mariadb:

secrets:
  DJANGO_SECRET_KEY:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true